src/kernel/standard/chdir.rs

```diff
@@ -1,77 +1,62 @@
-use crate::errors::Result;
-use crate::kernel::exit::SyscallExitResult;
+use nix::unistd::AccessFlags;
 
-pub fn enter() -> Result<()> {
-    Ok(())
+use crate::errors::*;
+use crate::filesystem::binding::Side;
+use crate::filesystem::Substitutor;
+use crate::process::tracee::Tracee;
+use crate::register::{Current, PtraceReader, SysArg, SysArg1, SysResult};
+
+pub fn enter(tracee: &mut Tracee) -> Result<()> {
+    let sys_num = tracee.regs.get_sys_num(Current);
+    let mut guest_path = if sys_num == sc::nr::CHDIR {
+        tracee.regs.get_sysarg_path(SysArg1)?
+    } else if sys_num == sc::nr::FCHDIR {
+        tracee.get_path_from_fd(
+            tracee.regs.get(Current, SysArg(SysArg1)) as i32,
+            Side::Guest,
+        )?
+    } else {
+        Err(Error::errno_with_msg(
+            Errno::ENOSYS,
+            format!(
+                "sysno should be CHDIR({}) or FCHDIR({}), but got {}",
```

Comment:
Unit test to purposefully return the following:
- [ ] CHDIR
- [ ] FCHDIR
- [ ] ???